hamocc/mo_parambgc_ini.F90

```diff
@@ -0,0 +1,686 @@
+    ! The order of initialization can matter due to interdependcies.
+    ! Then read the namelist and adjust rates to 'per time step'
+    ! Re-adjust dependent parameter values
+    ! Eventually write out the used parameters to the log file
+    !
+    implicit none      
+
+    INTEGER, intent(in) :: kpie,kpje
+
+    call ini_param_atm()           ! Initialize default atmospheric parameters
+    call ini_stoichiometry()       ! Initialize fixed stoichiometric parameters 
+    call ini_param_biol()          ! initialize biological parameters
+#ifdef AGG
+    call ini_aggregation()         ! Initialize aggregation module of Iris Kriest (no NML read thus far)
+#endif
+
+    call read_bgcnamelist()        ! read the BGCPARAMS namelist
+    call calc_param_atm()          ! calculate atmospheric parameters after updating parameters via nml
+    call ini_fields_atm(kpie,kpje) ! initialize atmospheric fields with (updated) parameter values 
+    call readjust_param()          ! potentially readjust namlist parameter-dependent parameters
+    call rates_2_timestep()        ! Converting rates from /d... to /dtb
+
+    call write_parambgc()          ! write out used parameters and calculate back rates from /dtb to /d..
+  end subroutine
+
+
+  !---------------------------------------------------------------------------------------------------------------------------------
+  subroutine ini_param_atm()
+    !
+    ! Atmospheric concentrations (atm_co2 is set via BGCNML namelist).
+    !
+    atm_o2      = 196800.
+    atm_n2      = 802000.
+
+#ifdef natDIC
+    atm_co2_nat = 284.32 ! CMIP6 pre-industrial reference
+#endif
+#ifdef BROMO
+    !For now use 3.4ppt from Hense and Quack (2009; Biogeosciences) NEED TO
+    !BE UPDATED WITH Ziska et al. (2013) climatology database
+    atm_bromo   = 3.4
+#endif
+#ifdef cisonew
+    ! set standard carbon isotope ratios
+    re1312      = 0.0112372
+    re14to      = 1.170e-12  ! Karlen et al. 1965 / Orr et al. 2017
+    ! set preindustr. d13c and bigd14C in atmosphere
+    prei13      = -6.5
+    prei14      =  0.
+#endif cisonew
+  end subroutine
+  
+
+  !---------------------------------------------------------------------------------------------------------------------------------
+  subroutine calc_param_atm()
+    !
+    ! AFTER having read the namelist:
+    ! calculate parameters for atmosphere from given parameters 
+    !
+#ifdef cisonew
+    beta13   = (prei13/1000.)+1.
+    alpha14  = 2.*(prei13+25.)
+    d14cat   = (prei14+alpha14)/(1.-alpha14/1000.)
+    ! calculate atm_c13 and atm_c14
+    atm_c13  = beta13*re1312*atm_co2/(1.+beta13*re1312)
+    d13C_atm = (((atm_c13/(atm_co2-atm_c13))/re1312)-1.)*1000.
+    ! absolute 14c concentration in preindustrial atmosphere
+    atm_c14  = ((d14cat/1000.)+1.)*re14to*atm_co2
+    ! factor for normalizing 14C tracers (~1e-12)
+    c14fac   = atm_c14/atm_co2
+#endif
+  end subroutine 
```

Comment:
A matter of personal taste, but I would prefer to have not too many small subroutines. I would place the calculations done here in other routines: The 3 first lines could go into  `ini_param_atm` and the rest into `ini_fields_atm` (`d13C_atm` is only used to print out the value, so it could go into `write_parambgc`)